packages/dds/spanner/src/spanner.ts

```diff
@@ -0,0 +1,156 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IFluidHandle, IFluidLoadable } from "@fluidframework/core-interfaces";
+import { assert } from "@fluidframework/core-utils";
+import {
+	IChannel,
+	IChannelAttributes,
+	IChannelFactory,
+	IChannelServices,
+	IFluidDataStoreRuntime,
+} from "@fluidframework/datastore-definitions";
+import {
+	ITelemetryContext,
+	ISummaryTreeWithStats,
+	IExperimentalIncrementalSummaryContext,
+	IGarbageCollectionData,
+} from "@fluidframework/runtime-definitions";
+import { ISequencedDocumentMessage } from "@fluidframework/protocol-definitions";
+
+import { SharedObject } from "@fluidframework/shared-object-base";
+import { SpannerHandle } from "./spannerHandle";
+import { SpannerChannelServices } from "./spannerChannelServices";
+
+interface IHotSwapOp {
+	type: "hotSwap";
+	oldAttributes: IChannelAttributes;
+	newAttributes: IChannelAttributes;
+}
+
+/**
+ * A channel that can swap Distributed Data Structures (DDS)
+ */
+export class Spanner<TOld extends SharedObject, TNew extends SharedObject> implements IChannel {
```

Comment:
I would have expected this to extend SharedObject.  Is there any motivation to deviate here?

Reply:
I think the biggest thing was getting it to compile while having a `get` accessor for the attributes. Maybe I was missing something, but `readonly` on the `attributes` property was throwing my compiler off. I was struggling trying to get the attributes to switch correctly so I could get the DDS to summarize correctly.